src/EFCore.PG/Design/Internal/NpgsqlCSharpRuntimeAnnotationCodeGenerator.cs

```diff
@@ -28,6 +29,106 @@ public class NpgsqlCSharpRuntimeAnnotationCodeGenerator : RelationalCSharpRuntim
     {
     }
 
+    /// <summary>
+    ///     This is an internal API that supports the Entity Framework Core infrastructure and not subject to
+    ///     the same compatibility standards as public APIs. It may be changed or removed without notice in
+    ///     any release. You should only use it directly in your code with extreme caution and knowing that
+    ///     doing so can result in application failures when updating to a new Entity Framework Core release.
+    /// </summary>
+    public override bool Create(
+        CoreTypeMapping typeMapping,
+        CSharpRuntimeAnnotationCodeGeneratorParameters parameters,
+        ValueComparer? valueComparer = null,
+        ValueComparer? keyValueComparer = null,
+        ValueComparer? providerValueComparer = null)
+    {
+        var result = base.Create(typeMapping, parameters, valueComparer, keyValueComparer, providerValueComparer);
```

Comment:
@AndriySvyryd here's some Npgsql-specific customizations to type mapping code generation in the compiled model, this causes an additional Clone() to be generated with the required stuff:

```c#
foo.TypeMapping = NpgsqlRangeTypeMapping.Default.Clone(
    comparer: new ValueComparer<NpgsqlRange<DateTime>>(
        (NpgsqlRange<DateTime> v1, NpgsqlRange<DateTime> v2) => v1.Equals(v2),
        (NpgsqlRange<DateTime> v) => v.GetHashCode(),
        (NpgsqlRange<DateTime> v) => v),
    keyComparer: new ValueComparer<NpgsqlRange<DateTime>>(
        (NpgsqlRange<DateTime> v1, NpgsqlRange<DateTime> v2) => v1.Equals(v2),
        (NpgsqlRange<DateTime> v) => v.GetHashCode(),
        (NpgsqlRange<DateTime> v) => v),
    providerValueComparer: new ValueComparer<NpgsqlRange<DateTime>>(
        (NpgsqlRange<DateTime> v1, NpgsqlRange<DateTime> v2) => v1.Equals(v2),
        (NpgsqlRange<DateTime> v) => v.GetHashCode(),
        (NpgsqlRange<DateTime> v) => v),
    mappingInfo: new RelationalTypeMappingInfo(
        storeTypeName: "tstzrange"),
    clrType: typeof(NpgsqlRange<DateTime>));
foo.TypeMapping = ((NpgsqlRangeTypeMapping)foo.TypeMapping).Clone(
    NpgsqlDbType.TimestampTzRange,
    NpgsqlTimestampTzTypeMapping.Default);
```

We may want to make this better in the future. For example, a type mapping could have an API which allows it to add a code fragment (or otherwise customize) its own initialization in the compiled model (a bit like how it has GenerateCodeLiteral, which generates a code fragment for instantiating a literal in C#).